core/api/src/app/authentication/totp.ts

```diff
@@ -90,14 +91,17 @@ const validateTotpRegistrationWithToken = async ({
   totpRegistrationId: TotpRegistrationId
   userId: UserId
 }): Promise<User | ApplicationError> => {
+  const res = await refreshToken(authToken)
+  if (res instanceof Error) return res
+
   const validation = await kratosValidateTotp({ authToken, totpCode, totpRegistrationId })
   if (validation instanceof Error) return validation
 
-  const res = await validateKratosToken(authToken)
-  if (res instanceof Error) return res
-  if (res.kratosUserId !== userId) return new AuthTokenUserIdMismatchError()
+  const res2 = await validateKratosToken(authToken)
```

Comment:
not sure to understand this... first refresh authToken and then validate?

Reply:
yes, there is notion of freshness for the login in kratos. certain critical action needs to require to have a recent login, or otherwise kratos will ask you to login again to make sure this is really you. 

the type of actions that requires this are things like changing password or adding TOTP. 

currently the way we have done our login it's not very straightforward to align with what kratos expect because of how we proxy the call to kratos, so we are just here forcing the refresh on the backend.

note: this was already done previously, I just refactor the code to have the refreshToken only if it was just not create prior.